cloudinit/sources/DataSourceAzure.py

```diff
@@ -634,6 +634,48 @@ def _get_data(self):
             self.ds_cfg['data_dir'], crawled_data['files'], dirmode=0o700)
         return True
 
+    @azure_ds_telemetry_reporter
+    def get_imds_data(
+            self,
+            fallback_nic,
+            retries,
+            md_type=metadata_type.compute):
+        LOG.info("Attempting IMDS api-version: %s", self.api_version)
+        if self.failed_desired_api_version:
+            return get_metadata_from_imds(
+                fallback_nic=fallback_nic,
+                retries=retries,
+                md_type=md_type,
+                api_version=self.api_version
+            )
+
+        try:
+            try:
+                return get_metadata_from_imds(
+                    fallback_nic=fallback_nic,
+                    retries=1,
```

Comment:
`retries=1` means try twice. `retries=0` means try once. Did you intend `retries=1`?